### PR TITLE
fn: new agent resource tracker metrics

### PR DESCRIPTION
**Pull request by skinowski**
New metrics for agent resource tracker: CpuUsed, CpuAvail,
MemUsed, MemAvail.

### CODE REVIEW

File: api/agent/agent.go [hunk 1]
```diff
@@ -832,11 +833,21 @@ func (a *agent) prepCold(ctx context.Context, call *call, tok ResourceToken, ch
 
 	call.containerState.UpdateState(ctx, ContainerStateIdle, call.slots)
 
-	slot := &coldSlot{cookie, tok, err}
+	closer := func() {
+		if cookie != nil {
+			cookie.Close(ctx)
```
Comment (rdallman) [inline api/agent/agent.go]: in general this isn't great, we're enclosing a stale context from creation into `cookie.Close` when it's called much later, perhaps with a different enclosing context at the time

Comment (rdallman) [reply api/agent/agent.go]: not just for timeout, but for stats/trace ordering it gets really weird too...

Comment (rdallman) [reply api/agent/agent.go]: it seems like this is cold only after looking, so this is ok really if hot doesn't have this shortcoming

Comment (skinowski) [reply api/agent/agent.go]: yes, cold only and n/a for hot.

**Pull request merged**